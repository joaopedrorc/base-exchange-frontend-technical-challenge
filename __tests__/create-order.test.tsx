import { render } from '@testing-library/react';
import CreateOrder from '@/app/create-order/page';

jest.mock('next/navigation', () => {
  return {
    useRouter: () => ({
      push: () => {},
    }),
  };
});

describe('CreateOrder page', () => {
  it('should render CreateOrder page', async () => {
    const ui = await CreateOrder(); // executa a função assíncrona
    const { getByText } = render(ui);

    expect(getByText('Votar')).toBeInTheDocument();
    expect(getByText('Crie uma Nova Ordem')).toBeInTheDocument();
  });

  it('renders CreateOrder unchanged', async () => {
    const ui = await CreateOrder();
    expect(ui).toMatchSnapshot();
  });
});
