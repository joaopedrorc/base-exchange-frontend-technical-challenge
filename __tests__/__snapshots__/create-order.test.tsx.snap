// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateOrder page renders CreateOrder unchanged 1`] = `
{
  "$$typeof": Symbol(react.transitional.element),
  "_owner": null,
  "_store": {},
  "key": null,
  "props": {
    "children": [
      {
        "$$typeof": Symbol(react.transitional.element),
        "_owner": null,
        "_store": {},
        "key": null,
        "props": {
          "children": {
            "$$typeof": Symbol(react.transitional.element),
            "_owner": null,
            "_store": {},
            "key": null,
            "props": {
              "children": {
                "$$typeof": Symbol(react.transitional.element),
                "_owner": null,
                "_store": {},
                "key": null,
                "props": {
                  "children": [
                    {
                      "$$typeof": Symbol(react.transitional.element),
                      "_owner": null,
                      "_store": {},
                      "key": null,
                      "props": {},
                      "type": {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                    },
                    {
                      "$$typeof": Symbol(react.transitional.element),
                      "_owner": null,
                      "_store": {},
                      "key": null,
                      "props": {
                        "children": "Votar",
                        "className": "text-sm",
                      },
                      "type": "span",
                    },
                  ],
                  "className": "flex flex-row items-center gap-1",
                  "href": "/",
                },
                "type": {
                  "$$typeof": Symbol(react.forward_ref),
                  "default": [Circular],
                  "render": [Function],
                  "useLinkStatus": [Function],
                },
              },
              "className": "bg-slate-800",
            },
            "type": [Function],
          },
          "className": "mb-4",
        },
        "type": "div",
      },
      {
        "$$typeof": Symbol(react.transitional.element),
        "_owner": null,
        "_store": {},
        "key": null,
        "props": {
          "children": [
            {
              "$$typeof": Symbol(react.transitional.element),
              "_owner": null,
              "_store": {},
              "key": null,
              "props": {
                "children": {
                  "$$typeof": Symbol(react.transitional.element),
                  "_owner": null,
                  "_store": {},
                  "key": null,
                  "props": {
                    "children": "Crie uma Nova Ordem",
                    "className": "text-slate-900",
                  },
                  "type": [Function],
                },
              },
              "type": [Function],
            },
            {
              "$$typeof": Symbol(react.transitional.element),
              "_owner": null,
              "_store": {},
              "key": null,
              "props": {
                "children": {
                  "$$typeof": Symbol(react.transitional.element),
                  "_owner": null,
                  "_store": {},
                  "key": null,
                  "props": {},
                  "type": [Function],
                },
                "className": "space-y-2 text-slate-800",
              },
              "type": [Function],
            },
          ],
          "className": "rounded-2xl border-slate-200 bg-slate-50 p-4 shadow-md",
        },
        "type": [Function],
      },
    ],
  },
  "type": "div",
}
`;
